src/filter.rs

```diff
@@ -218,17 +218,26 @@ impl Filter {
                 let filter_state = &mut filter_state[c];
 
                 src.foreach_sample_zipped(c, dest[dest_index..].iter_mut(), |src, dest| {
-                    filter_state[0] = (*src).to_sample::<f64>()
+                    let new_filter_state = (*src).to_sample::<f64>()
                         - a[1] * filter_state[1]
                         - a[2] * filter_state[2]
                         - a[3] * filter_state[3]
                         - a[4] * filter_state[4];
-                    *dest = b[0] * filter_state[0]
+
+                    let res = b[0] * filter_state[0]
                         + b[1] * filter_state[1]
                         + b[2] * filter_state[2]
                         + b[3] * filter_state[3]
                         + b[4] * filter_state[4];
 
+                    if new_filter_state.is_nan() || res.is_nan() {
```

Comment:
I wonder if there's a better way that's not so bad for performance. Maybe we can check outside the loop if the filter state contains a `NaN` and in that case reset everything?